traversal/procedure/GraphProcedure.java

```diff
@@ -183,10 +185,29 @@ public Builder() {
         }
 
         public GraphProcedure build() {
+            linearise();
             return new GraphProcedure(vertices.values().stream().sorted(comparing(ProcedureVertex::order))
                     .toArray(ProcedureVertex[]::new));
         }
 
+        private void linearise() {
+            Set<ProcedureVertex<?, ?>> orderedSet = new HashSet<>();
```

Comment:
shouldn't an `orderedSet` actually be an ordered set? where do we depend on the ordering? Either the naming is off or the type is wrong...

Reply:
It's meant to be a set of the ordered vertices. `orderedVertices` then.

Reply:
`visited`?